internal/scaffold/ansible/watches_test.go

```diff
@@ -0,0 +1,127 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ansible
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/util/fileutil"
+)
+
+func TestUpdateAnsibleWatchForResource(t *testing.T) {
+
+	watchFilePath := "./testdata/valid/watches.yaml"
+	if err := ioutil.WriteFile(watchFilePath, []byte(sampleRole), fileutil.DefaultFileMode); err != nil {
+		fmt.Printf("failed to update %v", err)
+	}
+
+	testCases := []struct {
+		name           string
+		r              *scaffold.Resource
+		absProjectPath string
+		expError       string
+	}{
+
+		{
+			//Valid watch
+			name: "Valid Watch",
+			r: &scaffold.Resource{
+				APIVersion: "app.example.com/v1alpha1",
+				Kind:       "App",
+				LowerKind:  "app",
+				FullGroup:  "app.example.com",
+				Version:    "v1alpha1",
+			},
+			absProjectPath: "./testdata/valid",
+			expError:       "",
+		},
+		{
+			//Invalid Watch
+			name: "Duplicate GVK",
+			r: &scaffold.Resource{
+				APIVersion: "app.example.com/v1alpha1",
+				Kind:       "App",
+				LowerKind:  "app",
+				FullGroup:  "app.example.com",
+				Version:    "v1alpha1",
+			},
+			absProjectPath: "./testdata/valid",
+			expError:       "duplicate GVK: app.example.com/v1alpha1, Kind=App",
+		},
+
+		{
+			//Invalid Watch
+			name: "Empty WatchFile",
+			r: &scaffold.Resource{
+				APIVersion: "app.example.com/v1alpha1",
+				Kind:       "App",
+				LowerKind:  "app",
+				FullGroup:  "app.example.com",
+				Version:    "v1alpha1",
+			},
+			absProjectPath: "./testdata/invalid/emptywatchfile",
+			expError:       "empty Watch File at: ./testdata/invalid/emptywatchfile",
+		},
+		{
+			//Invalid Watch
+			name: "Empty Directory",
+			r: &scaffold.Resource{
+				APIVersion: "app.example.com/v1alpha1",
+				Kind:       "App",
+				LowerKind:  "app",
+				FullGroup:  "app.example.com",
+				Version:    "v1alpha1",
+			},
+			absProjectPath: "./testdata",
+			expError: "failed to read watch manifest testdata/watches.yaml: " +
+				"open testdata/watches.yaml: no such file or directory",
+		},
+		{
+			//Invalid Watch
+			name: "Invalid Watch file",
+			r: &scaffold.Resource{
+				APIVersion: "app.example.com/v1alpha1",
+				Kind:       "App",
+				LowerKind:  "app",
+				FullGroup:  "app.example.com",
+				Version:    "v1alpha1",
+			},
+			absProjectPath: "./testdata/invalid/invalid_watch",
+			expError: "failed to unmarshal watch config yaml: unmarshal errors:" + "\n" +
+				"  line 2: cannot unmarshal !!map into []ansible.watchesYAML",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var result string
+			if err := UpdateAnsibleWatchForResource(tc.r, tc.absProjectPath); err != nil {
+				result = err.Error()
+			}
+			assert.Equal(t, tc.expError, result)
```

Comment:
Ditto from question in helm scaffold: can we also check the contents of the updated file?

Reply:
Please see my [comment](https://github.com/operator-framework/operator-sdk/pull/2703#discussion_r405509376) .